.github/workflows/publish.yml

```diff
@@ -61,8 +61,56 @@ jobs:
       - uses: actions/setup-go@v5
         with:
           go-version: 1.22.x
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'temurin'
       - name: Build binary
         run: GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }} make build
+      - name: Download jsign.jar
+        if: matrix.os == 'windows'
+        run: |
+          curl -L -o jsign.jar https://github.com/ebourg/jsign/releases/download/6.0/jsign-6.0.jar
+      - name: Verify jsign.jar
+        if: matrix.os == 'windows'
+        run: |
+          echo '05ca18d4ab7b8c2183289b5378d32860f0ea0f3bdab1f1b8cae5894fb225fa8a  jsign.jar' | sha256sum -c
+      - name: Prepare certificate chain
+        if: matrix.os == 'windows'
+        run: |
+          echo "${{ secrets.CERTIFICATE_CHAIN }}" | base64 --decode > codesign-chain.pem
+      - id: 'auth'
```

Comment:
Personally, I would move the auth and cloud sdk steps to immediately after the build. I would then merge all of these steps, such as downloading, validating the checksum, and signing into a single block, `if: ${{ startsWith(github.ref, 'refs/tags/') && matrix.os == 'windows' }}`

I don't think we should sign intermediate (main) builds, as we incur a cost for HSM usage.